semgrep/semgrep/target_manager.py

```diff
@@ -426,6 +426,23 @@ def files(self) -> FrozenSet[Path]:
         return self.files_from_filesystem()
 
 
+@lru_cache(maxsize=100_000)
+def get_shebang_line(path: Path) -> Optional[str]:
+    if not os.access(str(path), os.X_OK | os.R_OK):
+        return None
+
+    with path.open() as f:
+        return f.readline(MAX_CHARS_TO_READ_FOR_SHEBANG).rstrip()
+
+
+@lru_cache(maxsize=10_000)
```

Comment:
Question: why is this 10k and all the others are 100k?

Reply:
As I was profiling on a real life project, this function had an order of magnitude fewer calls than others. I tried to set all cache limits to be around 100x the unique calls I saw on returntocorp/semgrep, so that we cache everything but have a safety check in place in case there's an unexpected combinatorical explosion.

Reply:
Great, can we leave a comment to this effect?